src/module.c

```diff
@@ -1054,16 +1114,123 @@ int RMCreateSearchCommand(RedisModuleCtx *ctx, const char *name,
-    rc = REDISMODULE_ERR;
+    command = NULL;
   }
 
   if (!is_internal) {
     rm_free(categories);
   }
+  return command;
+}
+
+static int CreateSearchCommand(RedisModuleCtx *ctx, const SearchCommand *details) {
+  RedisModule_Log(ctx, "verbose", "Command %s, flags: %s, register: %d", details->name, details->flags, (int)details->shouldRegister);
+  if (!details->shouldRegister) {
+    return REDISMODULE_OK;
+  }
+
+  RedisModuleCommand *cmd = CreateCommandWithAcl(ctx, details->name, details->handler, details->flags, details->position, details->aclCategories);
+  if (!cmd) {
+    return REDISMODULE_ERR;
+  }
+  switch(details->selectedCallbackType) {
+    case SET_COMMAND_INFO:
+        if (details->callback.setCommandInfo(cmd) != REDISMODULE_OK) {
+            RedisModule_Log(ctx, "warning", "Could not set command info for command %s", details->name);
+            return REDISMODULE_ERR;
+        }
+        break;
+    case SUBSCRIBE_SUBCOMMANDS:
+        if (details->callback.subscribeSubCommands(ctx, cmd) != REDISMODULE_OK) {
+            RedisModule_Log(ctx, "warning", "Could not subscribe subcommands for command %s", details->name);
+            return REDISMODULE_ERR;
+        }
+        break;
+    case NONE:
+      RS_LOG_ASSERT_FMT(!details->callback.ptr, "Command registration callback was specified but the type of callback was not set for: %s", details->name);
+      break;
+  }
+  return REDISMODULE_OK;
+}
+
+/** A dummy command handler, for commands that are disabled when running the module in OSS
+ * clusters
+ * when it is not an internal OSS build. */
+int DisabledCommandHandler(RedisModuleCtx *ctx, RedisModuleString **argv, int argc) {
+  return RedisModule_ReplyWithError(ctx, "Module Disabled in Open Source Redis");
+}
+
+/** A wrapper function that safely checks whether we are running in OSS cluster when registering
+ * commands.
+ * If we are, and the module was not compiled for oss clusters, this wrapper will return a pointer
+ * to a dummy function disabling the actual handler.
+ *
+ * If we are running in RLEC or in a special OSS build - we simply return the original command.
+ *
+ * All coordinator handlers must be wrapped in this decorator.
+ */
+static RedisModuleCmdFunc SafeCmd(RedisModuleCmdFunc f) {
+  if (IsEnterprise() && clusterConfig.type != ClusterType_RedisLabs) {
+    /* If we are running inside OSS cluster and not built for oss, we return the dummy handler */
+    return DisabledCommandHandler;
+  }
+
+  /* Valid - we return the original function */
+  return f;
+}
+
+#define CONFIG_SUBCOMMANDS(command_, func_, readonly_)                                                                                                                           \
+  SubCommand subcommands[] = {                                                                                                                                      \
```

Comment:
I prefer to have subcommand as explicit variable outside the macro

Reply:
The problem is we need to fill fullName with two options.
The _FT. option and the FT. option.
It felt quicker than playing around in runtime with strings.